### PR TITLE
Add a criteria to dispatch flow by poll instead of containerization based on denylist of user.to.proxy

### PR DESCRIPTION
**Pull request by hqin2016**
tested on Holdem 6.

### CODE REVIEW

File: azkaban-common/src/main/java/azkaban/executor/container/ContainerProxyUserCriteria.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+/**
+ * Class for determining {@link DispatchMethod} based on deny list
+ */
+public class ContainerProxyUserCriteria {
+
+  private Set<String> denyList;
+
+  public ContainerProxyUserCriteria(final Props azkProps) {
+    this.denyList =
+        new HashSet<>(azkProps.getStringList(ContainerizedDispatchManagerProperties.CONTAINERIZED_PROXY_USER_DENYLIST));
```
Comment (jakhani) [inline azkaban-common/src/main/java/azkaban/executor/container/ContainerProxyUserCriteria.java:35]: For this, there should be a default value.

File: azkaban-common/src/main/java/azkaban/executor/container/ContainerProxyUserCriteria.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+  }
+
+  @VisibleForTesting
+  public void updateDenyList(final Set<String> denyList) { this.denyList = denyList; }
```
Comment (jakhani) [inline azkaban-common/src/main/java/azkaban/executor/container/ContainerProxyUserCriteria.java:39]: Do we need API options to update deny list without web server restart? If not, these methods are not required.

File: azkaban-common/src/main/java/azkaban/executor/container/ContainerProxyUserCriteria.java [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+  /**
+   * If the set of flow proxy users contains a proxy user on the deny list,
+   * then return DispatchMethod.POLL, else return DispatchMethod.CONTAINERIZED
+   *
+   * @param flow
+   * @return DispatchMethod
+   */
+  public DispatchMethod getDisPatchMethod(final ExecutableFlow flow) {
+    for (String proxyUser: flow.getProxyUsers()) {
```
Comment (jakhani) [inline azkaban-common/src/main/java/azkaban/executor/container/ContainerProxyUserCriteria.java:63]: Check if you want to use CollectionUtils.containsAny(someCollection1, someCollection2) or setA.stream().anyMatch(setB::contains).

File: azkaban-common/src/main/java/azkaban/executor/container/ContainerProxyUserCriteria.java (hunk 2 above)
Comment (hqin2016) [reply azkaban-common/src/main/java/azkaban/executor/container/ContainerProxyUserCriteria.java:39]: removed

File: azkaban-common/src/main/java/azkaban/executor/container/ContainerizedDispatchManager.java [hunk 4]
```diff
@@ -170,7 +176,11 @@ public DispatchMethod getDispatchMethod(final ExecutableFlow flow) {
     if (dispatchMethod != DispatchMethod.CONTAINERIZED) {
       return dispatchMethod;
     }
-    return this.containerJobTypeCriteria.getDispatchMethod(flow);
+    dispatchMethod = this.containerJobTypeCriteria.getDispatchMethod(flow);
+    if ( dispatchMethod != DispatchMethod.CONTAINERIZED) {
+      return dispatchMethod;
+    }
+    return this.containerProxyUserCriteria.getDisPatchMethod(flow);
```
Comment (jakhani) [inline azkaban-common/src/main/java/azkaban/executor/container/ContainerizedDispatchManager.java:183]: Nit: method name: getDis**p**atchMethod

File: azkaban-web-server/src/main/java/azkaban/webapp/servlet/ExecutorServlet.java [hunk 5]
```diff
@@ -1099,7 +1108,11 @@ private void ajaxRampActions(final HttpServletRequest req,
   UPDATE_ALLOW_LIST("updateAllowList"),
   APPEND_ALLOW_LIST("appendAllowList"),
   REMOVE_FROM_ALLOW_LIST("removeFromAllowList"),
-  UPDATE_RAMP_UP("updateRampUp");
+  UPDATE_RAMP_UP("updateRampUp"),
+  UPDATE_DENY_LIST("updateDenyList"),
```
Comment (jakhani) [inline azkaban-web-server/src/main/java/azkaban/webapp/servlet/ExecutorServlet.java:1112]: Do we need update?

Comment (hqin2016) [reply azkaban-web-server/src/main/java/azkaban/webapp/servlet/ExecutorServlet.java:1112]: removed update

Comment (hqin2016) [reply azkaban-web-server/src/main/java/azkaban/webapp/servlet/ExecutorServlet.java:1112]: I think removing updateRampUp is not in the scope of this PR. We can open a new pr to deal with this later.

**Review by jakhani — APPROVED** [review]

**Review by djaiswal83 — COMMENTED** [review]
Very nicely written code. Looks good to me.

**Pull request merged**